geonode/static/geonode/js/upload/upload.js

```diff
@@ -352,6 +352,10 @@ define(['underscore',
             buildFileInfo(_.groupBy(file_input.files, path.getName));
             displayFiles(file_queue);
         });
+        // Detect click on "Remove" link and update the file_queue
+        $(options.file_queue).on('click', 'a', function () {
+            displayFiles(file_queue);
+        });
```

Comment:
Could the jquery event selector use a class too?  Just `a` doesn't provide any semantic meaning.  Hopefully just minor change.